synapse/lib/msgpack.py

```diff
@@ -0,0 +1,58 @@
+import msgpack
+
+pakr = msgpack.Packer(use_bin_type=True, encoding='utf8')
+
+def en(item):
+    '''
+    Use msgpack to serialize a compatible python object.
+
+    Args:
+        item (obj): The object to serialize
+
+    Returns:
+        (bytes): The serialized bytes
+    '''
+    return pakr.pack(item)
+
+def un(byts):
+    '''
+    Use msgpack to de-serialize a python object.
+
+    Args:
+        byts (bytes): The bytes to de-serialize
+
+    Returns:
+        (obj): The de-serialized object
+    '''
+    return msgpack.loads(byts, use_list=False, encoding='utf8')
+
+class Unpk:
```

Comment:
Is the only reason for this class the fact that msgpack-python hasn't released a version with the .tell() API in place yet?

https://github.com/msgpack/msgpack-python/pull/227

Reply:
basically, yes.  it also conveniently wraps the OutOfData exception behavior ( which, in my opinion shouldn't raise, as it is not so much "exceptional" as "expected" for a streaming unpacker :) )